### ISSUE TITLE
[Bug] compiler breaks when using generics

### ISSUE THREAD

**Issue by RiddleNeon** [martin-pabst/Online-IDE-new-compiler#107]
There are no errors are shown, the play button is greyed out, it doesn't detect any errors anymore. 

<img width="2560" height="1290" alt="Image" src="https://github.com/user-attachments/assets/fe9fcec7-1876-4159-9174-118c1f3e8ebc" />

This error is repeatedly getting fired in the js console (about 10 times, then it stops): 

<img width="2554" height="188" alt="Image" src="https://github.com/user-attachments/assets/1ae5940c-7d59-47b1-8b81-b51953d5a4bf" />

as well as these warnings and logs in the beginning:

<img width="2553" height="123" alt="Image" src="https://github.com/user-attachments/assets/d81b1863-8adf-4451-b2bb-24fa9dc22465" />

duplicating the workspace does not work, other workspaces work, yesterday at about 18:00 PM CEST it still worked. 
If you go into a few other workspaces you haven't opened in a while it loads the run button (at least that's what i think it does, it shows a quick loading animation for a few frames) and go back into the problematic workspace the run button is green, however when you run it it runs only for 0-11 steps and stops immediately without any errors or whatsoever. it does not run anything you provided, even a simple print does not run. If you look at the js console after that it it you see that it re threw about 5 more of the type errors as well as an unreachable code warning at [GUILanguage.ts:212:48](https://www.online-ide.de/src/client/main/gui/language/GUILanguage.ts). additionally it also shows a slightly different type error from before:

```
TypeError: can't access property "push", e.catches is undefined
    canCastTo BinopCastCodeGenerator.ts:719
    searchMethod TermCodeGenerator.ts:1595
    compileMethodCall TermCodeGenerator.ts:1284
    compileTerm TermCodeGenerator.ts:87
    compileInitialValue StatementCodeGenerator.ts:1099
    compileLocalVariableDeclaration StatementCodeGenerator.ts:1027
    compileLocalVariableDeclarations StatementCodeGenerator.ts:1007
    compileStatementOrTerm StatementCodeGenerator.ts:58
    compileBlockNode StatementCodeGenerator.ts:576
    compileStatementOrTerm StatementCodeGenerator.ts:70
    compileMethodDeclaration InnerClassCodeGenerator.ts:596
    compileMethods InnerClassCodeGenerator.ts:723
    compileMethodsRecursively CodeGenerator.ts:181
    compileClassesEnumsAndInterfaces CodeGenerator.ts:55
    start CodeGenerator.ts:38
    compileIfDirty JavaCompiler.ts:149
    triggerCompile JavaCompiler.ts:274
    setTimeout handler*triggerCompile JavaCompiler.ts:268
    setWorkspaceActive ProjectExplorer.ts:887
    promise callback*setWorkspaceActive ProjectExplorer.ts:881
    nodeClickedCallback ProjectExplorer.ts:468
    onpointerup TreeviewNode.ts:336
```

# How to reproduce
import the following workspace and try running the program. 
[Workspace.json](https://github.com/user-attachments/files/29748291/Spiel.7.json)
[Spritesheet.zip](https://github.com/user-attachments/files/29748289/Spritesheet.7.zip)

# tested devices
mobile (safari, google browser app), linux (firefox)

**Comment by RiddleNeon** [comment]
Update: 
After inspecting the requests the site made I found 2 more potential issues: 
`GET wss://www.online-ide.de/servlet/pushWebsocket?csrfToken=mycsrftoken`
gives the status 101 (Switching Protocols) and was blocked. it gives the error message `NS_ERROR_WEBSOCKET_CONNECTION_REFUSED`.

Also
`POST https://www.online-ide.de/updateGuiState` 
gives a 404.

**Comment by RiddleNeon** [comment]
Okay so i kind of found the Issue and its really weird. I found the latest backup when it worked and compared it with the next version. i started copying stuff from the newer version until it broke and the responsible class was my PriorityQueue implementation. 
To be specific it is the generics it uses. I dont know what caused it to break but it was declared as 
`class PriorityQueue<T extends Comparable>` and that broke it. as soon as i removed that and replaced every T with Comparable the compiler worked again. However theres another bug that prevents me from doing that and i dont know what do do about that. 
This is my QueueNode:
```java
class QueueNode implements Comparable<QueueNode> {

   FlowFieldCell cell;
   double priority;

   public QueueNode(FlowFieldCell cell, double priority) {
      this.cell = cell;
      this.priority = priority;
   }

   public int compareTo(QueueNode o) {
      return priority == o.priority ? 0 : (priority > o.priority ? 1 : -1);
   }
}
```

However doing anything like this: 
```java
class ComparableClass implements Comparable {...}
private static void test(Comparable c) {}

ComparableClass testObject = new ComparableClass(testObject, 1);
test(testObject);
```
Does not work. it doesnt find the method since it requires you to pass in a comparable and even though a queue node implements a Comparable it doesnt allow it. When casting the QueueNode it says the cast is redundant and gives the same error anyway. Am i doing something wrong here?

Here is the partially working workspace:

[game.json](https://github.com/user-attachments/files/29750626/gameFixed.json)

**Comment by martin-pabst** [comment]
I fixed the main error. It's cause was a bug introduced when i extended the compiler to make `Arrays.asList` work. 
Please excuse the mistake!

As for the requests
`GET wss://www.online-ide.de/servlet/pushWebsocket?csrfToken=mycsrftoken` and `POST https://www.online-ide.de/updateGuiState` 
I'll look into this matter this weekend.

**Comment by martin-pabst** [comment]
I looked into the requests
  * a) `GET wss://www.online-ide.de/servlet/pushWebsocket?csrfToken=mycsrftoken` and 
  * b) `POST https://www.online-ide.de/updateGuiState` 

a) The Protocol switch (101) to websocket-protocol is correct. Unforunately websocket protocol is blocked in many schools due to students playing online browser games (which have fallback options by now...). Do you happen do be behind a school firewall/webfilter? 
When using the online-ide at home or on my office-pc at school it has no problem using websocket communication, so there seem to be no server side problems. If the online-ide encounters problems with websocket, it falls back to https long polling.
On your browser's console you should get messages like 'Opening websocket strategy' or 'Opening long polling strategy' or 'Websocket has been closed, code: 1005'. What do they read?

b) Thank you for pointing out this one! It should be `https://www.online-ide.de/servlet/updateGuiState`. I found the caller and fixed the error.

**Comment by RiddleNeon** [comment]
hm, I did get that error at home. But i didn't encounter any issues so i think trying to fix that isn't that important
